discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -176,12 +178,14 @@ def parse_sol_play_transaction(session, solana_client, tx_sig):
                             signature=tx_sig,
                         )
                     )
-                    challenge_bus.dispatch(
-                        ChallengeEvent.track_listen,
-                        tx_slot,
-                        user_id,
-                        {"created_at": created_at.timestamp()},
-                    )
+
+                    if user_id is not None:
```

Comment:
maybe comment on why this was super important 😂  (or why user_id might be none here for anon plays)